Use dateinterval in periodservice

BacLuc opened the pull request.

BacLuc committed:
PeriodTest: add tests for resize Period

Also add EDateInterval helper class because
the php DateInterval class ist not as convenient as
it should be.

Follow up to: #1078

BacLuc committed:
PeriodService: use php to calculate minutes between dates

The php DateInterval is not very convenient, but better
than doing it by ourselves at have numbers like 60 and 24 in our code.

Follow up to: #1078

pmattmann reviewed (commented):
Cool, danke für die Move-ScheduleEntry-Tests.

Magst du für `EDateInterval` auch Test schreiben?

BacLuc committed:
backend/EDateInterval: add test

Improve ofHours and ofDays that they support negative
integers too.
Implement format to nearly comply to liskovs substitution principle.
I could not set the days property after instantiating the
DateTime or the EDateTime object.

BacLuc commented on backend/module/eCampLib/test/Types/EDateIntervalTest.php:
```diff
@@ -0,0 +1,61 @@
+use eCamp\Lib\Types\EDateInterval;
+use PHPUnit\Framework\TestCase;
+
+/**
```
php-cs-fixer will das jetzt, hoffentlich kommt bald das update auf 2.18.5, das diese kleinen Bugs fixed.

carlobeltrame commented on backend/module/eCampLib/src/Types/EDateInterval.php:
```diff
@@ -0,0 +1,56 @@
+        return $dateInterval;
+    }
+
+    public static function totalSecondsOf(\DateInterval $interval): int {
```
Gibt es keine native Methode dafür? Wieso nicht? Kann das zu Problemen bei Schalttagen oder Schaltsekunden führen? Könntest du statt new DateTime ein fixes Datum nehmen damit diese Methode wirklich deterministisch wird?

BacLuc replied:
Ich habe leider nichts gefunden, siehe https://www.php.net/manual/en/dateinterval.format.php
Schalttage/Schaltsekunden sollten kein Problem sein, da das Interval gleich bleiben sollte.
Ein fixes Datum kann ich nehmen.

carlobeltrame reviewed (approved):
Merci!
